crates/bmp-pkt/src/wire/deserializer/mod.rs

```diff
@@ -51,16 +52,29 @@ pub enum BmpMessageParsingError {
     BmpMessageValueError(#[from_located(module = "self")] BmpMessageValueParsingError),
 }
 
-impl<'a>
-    ReadablePduWithOneInput<
-        'a,
-        &mut HashMap<PeerKey, BgpParsingContext>,
-        LocatedBmpMessageParsingError<'a>,
-    > for BmpMessage
+#[derive(Debug, Default)]
+pub struct BmpParsingContext(pub HashMap<PeerKey, BgpParsingContext>);
```

Comment:
Here you expose to APIs to modify the BmpParsingContext, that could be confusing.
1. First API, not clean since it exposes internal data structure: `ctx.0.remove`
2. the second relying on the `Deref` implementation: `ctx.remove`

Better to change `pub HashMap<PeerKey, BgpParsingContext` to `HashMap<PeerKey, BgpParsingContext` without pub and just keep one API

Reply:
true, didn't notice while doing it. let's keep the deref api